hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java

```diff
@@ -935,6 +936,16 @@ void freeSpace(final String why) {
     }
     try {
       freeInProgress = true;
+
+      // Check the list of files to determine the cold files which can be readily evicted.
+      Set<String> coldFiles =
+        DataTieringManager.getInstance().getColdDataFiles(backingMap.keySet());
+      if (coldFiles != null) {
+        for(String fileName : coldFiles) {
+          evictBlocksByHfileName(fileName);
+        }
+      }
+
```

Comment:
We can do like this here, but then can we think if we can modify DataTieringManager somehow to track when we have no CF with TIME_RANGE type? That way we could avoid this extra loops through the whole set of block keys unnecessarily.

Alternatively, we could follow the BucketEntryGroup logic starting from #987. We could define a COLD priority group and add that as the first group in the priority queue. That way we don't need this extra loop over the whole block sets to find out which blocks are cold. We would leverage the already existing loop from line #996 for that.